core/lib/spree/core/engine.rb

```diff
@@ -87,6 +87,7 @@ class Engine < ::Rails::Engine
 
       config.after_initialize do
         Spree::Config.check_load_defaults_called('Spree::Config')
+        Spree::Config.static_model_preferences.validate!
```

Comment:
What if you set the preference value after the app is initialized? Do you need to call `validate!` manually as well?

Reply:
After the app is initialized you could change the model configuration (e.g. adding or removing a preference) not the static preference value, but I regarded that as infrequent.

If we want to do it at every request in development we could move it to a `to_prepare` block, would you prefer that?

Reply:
You are right, they live in memory and we can't change their value at runtime. At least, not easily and it's not an expected usage.